app/Bus/Handlers/Traits/StoresMeta.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+/*
+ * This file is part of Cachet.
+ *
+ * (c) Alt Three Services Limited
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace CachetHQ\Cachet\Bus\Handlers\Traits;
+
+use CachetHQ\Cachet\Models\Meta;
+
+trait StoresMeta
+{
+    /**
+     * Stores all Meta values of a model.
+     *
+     * @param $meta
+     * @param $type
+     * @param $id
+     */
+    public function storeMeta($meta, $type, $id)
+    {
+        // Validation required instead of type hinting because it could be passed as false or NULL
+        if (!is_array($meta)) {
+            return;
+        }
+
+        foreach ($meta as $key => $value) {
+            if (empty($value)) {
```

Comment:
Hmm, what if we did just want a key but with an empty value?

Reply:
The database `value` column is not nullable. I could loosen up this check by allowing empty strings. But this would just pollute the database in my opinion.

Reply:
I just realized it's impossible to remove your custom meta attributes this way. So allow me to rewrite it a bit:

- If the value is considered emtpy, the row will be removed 
- If the value is not empty, the row will be created or updated

Reply:
Makes sense.